### PR TITLE
feat: [SFEQS-1303]  Send billing events to data-lake via event-hubs

### PR DESCRIPTION
**Pull request by grausof**
This PR adds event management. In particular it adds the sending of billing events on an event-hub. Please refer to the Billing documentation on confluence.

Depends on https://github.com/pagopa/io-infra/pull/396

#### List of Changes
Added `event` type to package `io-sign`
Added `azure/infra/event-hubs` integration on `io-func-sign-issuer`
Added dispatch of the billing event in the `mark-request-signed` use-case
Added `EventHubsConnectionString` environment variable

#### How Has This Been Tested?
Tested locally with cloud resources

#### Checklist:
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/io-sign/src/event.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { IsoDateFromString } from "@pagopa/ts-commons/lib/dates";
```
Comment (gunzip) [inline packages/io-sign/src/event.ts:1]: what about having this a a directory containing the single `<event>.ts` instead?

File: packages/io-sign/src/event.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import { Id, newId } from "./id";
+import {
+  SignatureRequestId,
+  SignatureRequestSigned,
+} from "./signature-request";
+
+const EventId = Id;
+
+// We currently only have a free testing plan and a paid standard plan
+export const PricingPlan = t.union([t.literal("FREE"), t.literal("DEFAULT")]);
```
Comment (gunzip) [inline packages/io-sign/src/event.ts:13]: Consider the usage of `t.keyOf`

https://github.com/gcanti/io-ts/tree/1.x#union-of-string-literals

**Review by gunzip — APPROVED** [review]

File: apps/io-func-sign-issuer/src/app/use-cases/mark-request-signed.ts [hunk 3]
```diff
@@ -24,5 +26,19 @@ export const makeMarkRequestAsSigned =
...
+      TE.chain(() =>
+        pipe(
+          request,
+          /*
+          The plan to use varies according to the environment used by the issuer.
+          If it is in a test environment the free plan should be used otherwise standard.
+          */
+          createBillingEvent(
+            request.issuerEnvironment === "TEST" ? "FREE" : "DEFAULT",
+            request.issuerId
```
Comment (lucacavallaro) [inline apps/io-func-sign-issuer/src/app/use-cases/mark-request-signed.ts:39]: imho this logic should be placed inside `createBillingEvent`

or maybe you can introduce a new function called `createBillingEventFromIssuer` that compose `createBillingEvent` with this logic

File: packages/io-sign/src/event.ts (hunk 2 above)
Comment (lucacavallaro) [inline packages/io-sign/src/event.ts:13]: use `t.keyof` instead of `t.union` https://github.com/gcanti/io-ts/blob/master/index.md#union-of-string-literals

**Review by lucacavallaro — CHANGES_REQUESTED** [review]

File: packages/io-sign/src/event.ts (hunk 1 above)
Comment (grausof) [reply packages/io-sign/src/event.ts:1]: For all other entities it was done this way inside `io-sign`. If we change for event we should also change for other entities.

Comment (lucacavallaro) [reply packages/io-sign/src/event.ts:1]: @gunzip Why do you recommend making a directory? Personally, I have no issues with dividing up our domain's directories if doing so makes the project simpler to navigate.

I believe that `event.ts` is now too short to be split up into several files; but, as this module grows, we may decide to reevaluate this in the future.

File: apps/io-func-sign-issuer/src/app/use-cases/mark-request-signed.ts (hunk 3 above)
Comment (grausof) [reply apps/io-func-sign-issuer/src/app/use-cases/mark-request-signed.ts]: I created a new function `createBillingEventFromIssuer` but I had to add `getIssuerById` in `infra/azure/cosmos/issuer`

Comment (grausof) [reply apps/io-func-sign-issuer/src/app/use-cases/mark-request-signed.ts]: I moved this check inside `createBillingEvent`

**Review by lucacavallaro — APPROVED** [review]

**Pull request merged**